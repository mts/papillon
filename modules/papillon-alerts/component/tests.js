import React from 'react';
import { shallow } from 'enzyme';

import {
  flashAlertDefault,
  multipleParagraphs,
  flashMessages,
  flashWarn,
  flashError
} from './flavors';

const renderShallow = () => {
  return shallow(flashAlertDefault);
};

describe('<FlashAlert />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of FlashAlert', () => {
      wrapper = renderShallow();
      expect(wrapper.find('FlashAlert')).toBeTruthy();
    });
  });

  describe('Snaphot', () => {
    test('must match flashAlertDefault', () => {
      expect(global.renderToJSON(flashAlertDefault)).toMatchSnapshot();
    });

    test('must match multipleParagraphs', () => {
      expect(global.renderToJSON(multipleParagraphs)).toMatchSnapshot();
    });

    test('must match flashMessages', () => {
      expect(global.renderToJSON(flashMessages)).toMatchSnapshot();
    });

    test('must match flashWarn', () => {
      expect(global.renderToJSON(flashWarn)).toMatchSnapshot();
    });

    test('must match flashError', () => {
      expect(global.renderToJSON(flashError)).toMatchSnapshot();
    });
  });
});
