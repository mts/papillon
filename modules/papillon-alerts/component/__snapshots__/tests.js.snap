// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashAlert /> Snaphot must match flashAlertDefault 1`] = `
<div
  className="p-4"
>
  <div
    className="flash"
  >
    Flash message goes here.
  </div>
</div>
`;

exports[`<FlashAlert /> Snaphot must match flashError 1`] = `
<div
  className="p-4"
>
  <div
    className="flash flash-error"
  >
    Flash message goes here.
  </div>
</div>
`;

exports[`<FlashAlert /> Snaphot must match flashMessages 1`] = `
<div
  className="p-4"
>
  <div
    className="flash-messages"
  >
    <div
      className="flash"
    >
      Flash message goes here.
    </div>
  </div>
</div>
`;

exports[`<FlashAlert /> Snaphot must match flashWarn 1`] = `
<div
  className="p-4"
>
  <div
    className="flash flash-warn"
  >
    Flash message goes here.
  </div>
</div>
`;

exports[`<FlashAlert /> Snaphot must match multipleParagraphs 1`] = `
<div
  className="p-4"
>
  <div
    className="flash"
  >
    <p>
      This is a longer flash message in it's own paragraph. It ends up looking something like this. If we keep adding more text, it'll eventually wrap to a new line.
    </p>
    <p>
      And this is another paragraph.
    </p>
  </div>
</div>
`;
